doppler/resource_service_account_identity.go

```diff
@@ -0,0 +1,245 @@
+package doppler
+
+import (
+	"context"
+	"errors"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceServiceAccountIdentity() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceServiceAccountIdentityCreate,
+		ReadContext:   resourceServiceAccountIdentityRead,
+		UpdateContext: resourceServiceAccountIdentityUpdate,
+		DeleteContext: resourceServiceAccountIdentityDelete,
+		Schema: map[string]*schema.Schema{
+			"service_account_slug": {
+				Description: "Slug of the service account",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"slug": {
+				Description: "Slug of the service account identity",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"name": {
+				Description: "The display name of the service account identity",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"ttl_seconds": {
+				Description: "The amount of time, in seconds, that auth tokens for this identity will be valid",
+				Type:        schema.TypeInt,
+				Required:    true,
+			},
+			"config_oidc": {
+				Description: "The OIDC configuration for the identity",
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				MinItems:    1,
+				Required:    true,
+				Elem:        &resourceServiceAccountIdentityConfigOidc,
+			},
+		},
+	}
+}
+
+var resourceServiceAccountIdentityConfigOidc = schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"discovery_url": {
+			Description: "The public URL of the OpenID discovery service",
+			Type:        schema.TypeString,
+			Required:    true,
+		},
+		"claims_type": {
+			Description: "If \"wildcard\", wildcard characters will be expanded during claims validation. Defaults to \"exact\"",
```

Comment:
Might be worth explicitly noting that the only two valid options here are `wildcard` and `exact`. It's kind of implied by the phrasing now, but an explicit note would be good since this is the description shown in the TF provider docs.